DEVELOPER_GUIDE.md

```diff
@@ -102,3 +103,20 @@ The Distribution Download plugin downloads the latest version of OpenSearch by d
 ```
 ./gradlew integTest -PcustomDistributionUrl="https://ci.opensearch.org/ci/dbc/bundle-build/1.2.0/1127/linux/x64/dist/opensearch-1.2.0-linux-x64.tar.gz"
 ```
+
+## Changelog
+
+AD maintains version specific changelog by enforcing a change to the ongoing [CHANGELOG](CHANGELOG.md) file adhering to the [Keep A Changelog](https://keepachangelog.com/en/1.1.0/) format.
+
+Briefly, the changes are curated by version, with the changes to the main branch added chronologically to `Unreleased` version. Further, each version has corresponding sections which list out the category of the change - `Added`, `Changed`, `Deprecated`, `Removed`, `Fixed`, `Security`.
+
+#### How to add my changes to [CHANGELOG](CHANGELOG.md)?
+
+As a contributor, you must ensure that every pull request has the changes listed out within the corresponding version and appropriate section of [CHANGELOG](CHANGELOG.md) file.
+
+Adding in the change is two step process -
+1. Add your changes to the corresponding section within the CHANGELOG file with dummy pull request information, publish the PR
+
+`Your change here ([#PR_NUMBER](PR_URL))`
+
+2. Update the entry for your change in [`CHANGELOG.md`](CHANGELOG.md) and make sure that you reference the pull request there.
```

Comment:
Can you add your example PR link?

Also, can you add the following:

Next time while creating the release notes all the entries can be copied directly to the release notes and can be deleted from either 2.x or 3.0 section as a clean up. Example: https://github.com/opensearch-project/flow-framework/pull/1036/files

If for changes we don't need to add an entry for such PRs we can add skip-changelog label, that would pass the changelog workflow

Reply:
Pushed the change